clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java

```diff
@@ -218,7 +220,8 @@ public DaVinciBackend(
           cacheBackend,
           true,
           // TODO: consider how/if a repair task would be valid for Davinci users?
-          null);
+          null,
+          new PubSubClientsFactory(new ApacheKafkaProducerAdapterFactory()));
```

Comment:
Here we directly pass the `PubSubClientFactory` with Kafka-wise implementation, so in the integration test it can only use Kafka-wise clients. If we want to test with Xinfra-wise implementation, DaVinci related test might not work here. This might be also true to `IsolatedIngestionServer.java` related integration test. We might need to come up some way to let the test get the `PubSubClientFactory` from either configs or parameters.

Reply:
Yeah, that's right. To be able to DVC and II will have Kafka clients as default for now. This will be addressed in PART-II of this work.